zproject/test_extra_settings.py

```diff
@@ -0,0 +1,232 @@
+else:
+    # This covers the backend test suite case
+    TORNADO_SERVER = None
+    CAMO_URI = 'https://external-content.zulipcdn.net/external_content/'
+    CAMO_KEY = 'dummy'
+
+CASPER_TESTS = "CASPER_TESTS" in os.environ
+if CASPER_TESTS:
+    # Disable search pills prototype for production use
+    SEARCH_PILLS_ENABLED = False
+
+if "RUNNING_OPENAPI_CURL_TEST" in os.environ:
+    RUNNING_OPENAPI_CURL_TEST = True
+
+if "GENERATE_STRIPE_FIXTURES" in os.environ:
+    GENERATE_STRIPE_FIXTURES = True
+
+# Decrease the get_updates timeout to 1 second.
+# This allows CasperJS to proceed quickly to the next test step.
+POLL_TIMEOUT = 1000
+
+# Don't use the real message log for tests
+EVENT_LOG_DIR = '/tmp/zulip-test-event-log'
+
+# Stores the messages in `django.core.mail.outbox` rather than sending them.
+EMAIL_BACKEND = 'django.core.mail.backends.locmem.EmailBackend'
+
+# The test suite uses EmailAuthBackend
+AUTHENTICATION_BACKENDS += ('zproject.backends.EmailAuthBackend',)
+
+# Configure Google Oauth2
+GOOGLE_OAUTH2_CLIENT_ID = "test_client_id"
+
+# Makes testing LDAP backend require less mocking
+AUTH_LDAP_ALWAYS_UPDATE_USER = False
+AUTH_LDAP_USER_SEARCH = LDAPSearch("ou=users,dc=zulip,dc=com",
+                                   ldap.SCOPE_ONELEVEL, "(uid=%(user)s)")
+AUTH_LDAP_USERNAME_ATTR = "uid"
+AUTH_LDAP_REVERSE_EMAIL_SEARCH = LDAPSearch("ou=users,dc=zulip,dc=com",
+                                            ldap.SCOPE_ONELEVEL,
+                                            "(mail=%(email)s)")
+
+TEST_SUITE = True
+RATE_LIMITING = False
+RATE_LIMITING_AUTHENTICATE = False
+# Don't use rabbitmq from the test suite -- the user_profile_ids for
+# any generated queue elements won't match those being used by the
+# real app.
+USING_RABBITMQ = False
+
+# Disable the tutorial because it confuses the client tests.
+TUTORIAL_ENABLED = False
+
+# Disable use of memcached for caching
+CACHES['database'] = {
+    'BACKEND': 'django.core.cache.backends.dummy.DummyCache',
+    'LOCATION': 'zulip-database-test-cache',
+    'TIMEOUT': 3600,
+    'CONN_MAX_AGE': 600,
+    'OPTIONS': {
+        'MAX_ENTRIES': 100000
+    }
+}
+
+# Disable caching on sessions to make query counts consistent
+SESSION_ENGINE = "django.contrib.sessions.backends.db"
+
+# Use production config from Webpack in tests
+if CASPER_TESTS:
+    WEBPACK_FILE = 'webpack-stats-production.json'
+else:
+    WEBPACK_FILE = os.path.join('var', 'webpack-stats-test.json')
+WEBPACK_LOADER['DEFAULT']['BUNDLE_DIR_NAME'] = 'webpack-bundles/'
+WEBPACK_LOADER['DEFAULT']['STATS_FILE'] = os.path.join(DEPLOY_ROOT, WEBPACK_FILE)
+
+# Don't auto-restart Tornado server during automated tests
+AUTORELOAD = False
+
+if not CASPER_TESTS:
+    # Use local memory cache for backend tests.
+    CACHES['default'] = {
+        'BACKEND': 'django.core.cache.backends.locmem.LocMemCache'
+    }
+
+    def set_loglevel(logger_name: str, level: str) -> None:
```

Comment:
I assume this pre-dates this change, but is there a reason that we exclude all of these `set_loglevel` calls for the `CASPER_TESTS` case?

Reply:
Yeah this change should just be moving existing code around (try `git show --color-moved`). The conditional goes back to commit 2dc553df0a8f2211cecef9336485e0e73afa661b, which claims: “Previously the casper server.log files basically only had tracebacks; this should help a lot with debuggability of Casper issues.”